### ISSUE TITLE
False positive non-iterable error for Enum instances created with functional API

### ISSUE THREAD

**Issue by ddoskind** [PyCQA/pylint#1546]
Somewhat similar to #1539.

### Steps to reproduce

Run pylint on the following code:
````python
from enum import Enum

Animal = Enum('Animal', 'ant bee cat dog')

for animal in Animal:
    print(animal)
````

### Current behavior

pylint emits an error concerning the for loop:

`E:  5,14: Non-iterable value Animal is used in an iterating context (not-an-iterable)`

### Expected behavior
pylint recognizes that Enums can be iterated over even when they are created using the Functional API. 

### pylint --version output
````
$ pylint --version
No config file found, using default configuration
pylint 1.7.1, 
astroid 1.5.3
Python 3.6.1 (default, Apr  4 2017, 09:40:21) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
````
